core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -2700,6 +2700,102 @@ class WorkspaceServiceSpec
       workspace.bucketName should startWith(s"${services.workspaceServiceConfig.workspaceBucketNamePrefix}-secure")
   }
 
+  it should "clone the WSM stub workspace if it exists" in withTestDataServices { services =>
+    val baseWorkspace = testData.workspace
+    val newWorkspaceName = "cloned_space"
+    val workspaceRequest = WorkspaceRequest(testData.testProject1Name.value, newWorkspaceName, Map.empty)
```

Comment:
How do we know in this test case that the stub workspace exists?

Reply:
The WSM dao used in this test returns a successful `CloneWorkspaceResult` instead of throwing a 404 which is interpreted as the workspace existing. The verify at the end of the test ensures that the cloneWorkspace request is issued.